### ISSUE TITLE
Standardize all line endings to be the same format

### ISSUE THREAD

**Issue by Brett208** [OutpostUniverse/OP2Utility#114]
It looks like I managed to switch many of the line endings from CRLF to just LF on this commit:

SHA-1: f6f02fdd61dc88ade600450c990e859626642dec
Moved project folder to the same folder as solution 
9/30/2017

I would propose we update all line endings to be the same. I would probably go with CRLF but I guess this is a cross platform project now.

**Comment by DanRStevens** [comment]
We should standardize.

I think the general trend is towards LF these days. All the GitHub docs seem to suggest settings that favour LF stored in the repo. MacOS switched to LF (previously it was CR). Only Windows and [Teletype machines](https://en.wikipedia.org/wiki/Teleprinter) still us CRLF (and yes, that's where it comes from, it's that old, check the section on "Control Characters").

With the increase in cross platform development, I see more and more programmers using LF, even on Windows, particularly for non Visual Studio languages.

I'm not sure what the case is like for Visual Studio, though it seems you were using LF for a long time and didn't even notice? I was under the impression the tooling is still a bit weak there on this issue. I've heard there are various plugins to support line ending settings and conversions. I also heard Visual Studio project and solution files have to use CRLF. Fortunately, those files are unlikely to ever be used or edited on Linux.

I think line endings are more of an editor concern than a version control concern. If we could solve this at the editor level, that would be best.

I know a lot of newer common code editors have good support for both line endings (e.g. Notepad++, Atom, VSCode, many others). In particular, they have an option to set the default EOL for new files, and the ability to detect and maintain a consistent EOL for existing files. Heck, even Notepad was updated to support other line endings. (It happened just this year, and might not be part of the mainstream release yet). Conversion options are also commonly supported.

Since Visual Studio is a big part of development here, we'll need to consider what Visual Studio supports. If it has decent support for LF, I'd like to go with LF. If not, I think the other tools are capable of going the other way.

Most of the time, I don't even notice which line endings we are using. At least not until a sensitive tool fails on it (bash scripts for instance, which only support LF), or they get mass updated. Speaking of which, we should avoid mass updates, such as moving files, or line ending conversions, while there are open unmerged branches.

**Comment by Brett208** [comment]
https://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/

We can try switching everything to LF line endings.

There may be some unkown compatility problems if an LF text document is archived into a volume file and then that volume is loaded by Outpost 2. This could be an issue if we start storing some sort of txt documents in the repository for use during automated testing.

-Brett

**Comment by DanRStevens** [comment]
Heh, I came across that article too in my reading. The `core.safecrlf = true` setting looks like it can be useful to prevent issues with accidentally modifying binary files. Might even impact mixed mode files, should we manage to introduce both line endings to the same file.

You raise an interesting point, in that Git might modify a version controlled precursor file before packing, so that's an interesting thought. The archive files themselves won't be modified by Git, since they are binary files.

I don't imagine we'll run into issues with automated testing. We'd likely be running automated tests in a platform independent manner, outside of the game, testing the code against itself and against existing data. That should negate line ending concerns, since the tests would be wholly contained within one environment for each run.

Perhaps automated packaging or deployment might encounter an issue with packing an LF text document. Though if we ever managed to do that, I think we'd learn pretty quickly. :wink: 

Do you think you'll try to solve the Windows development side of this with Git settings, or with editor configuration?

**Comment by Brett208** [comment]
I've been playing with auto CLRF setting in Git. It is working well as advertised. I had been using it before without incident until we attempted to move the sourcecode, even though I didn't know how it worked. I don't know what made it break down at that point. Now that I'm more familiar with it though, I'm hoping to catch the problem and fix it earlier if it crops up again.

So, I'm using CLRF when developing and letting Git convert back. Now that I know what it looks like when it breaks down, we should be able to catch it before anything gets merged in incorrectly. I suspect it will work mostly flawlessly in the future?

To recap weird behavior to watch for: Visual Studio project files will use LF line endings in the REPO, which doesn't make sense on Windows (but Git will compensate by converting them to CRLF on Windows). Text documents that may be packaged for testing purposes and later loaded into Outpost 2 could experience problems because they contain LF endings which we don't know how Outpost 2 would handle from a checksum perspective and from properly loading the data.

-Brett

**Comment by DanRStevens** [comment]
Sounds like this had been sorted out now, and we have a working plan moving forward.
